pkg/controller/networkpolicy/validate.go

```diff
@@ -145,21 +145,19 @@ func (v *NetworkPolicyValidator) validateAntreaPolicy(op admv1.Operation, tier s
 }
 
 // validateRuleName validates if the name of each rule is unique within a policy
-func (v *NetworkPolicyValidator) validateRuleName(ingress, egress []secv1alpha1.Rule) bool {
+func (v *NetworkPolicyValidator) validateRuleName(rules []secv1alpha1.Rule) (bool, string) {
 	uniqueRuleName := sets.NewString()
-	isUnique := func(rules []secv1alpha1.Rule) bool {
-		for _, rule := range rules {
-			if uniqueRuleName.Has(rule.Name) {
-				return false
-			}
-			if rule.Name != "" {
-				uniqueRuleName.Insert(rule.Name)
-			}
+	for _, rule := range rules {
+		// If a rule doesn't have a name, it means that
+		// mutatingWebhook skip its generation due to illegal input
+		if rule.Name == "" {
+			return false, "policy has duplicate rules, or collision occurred in generated rule names"
+		}
+		if uniqueRuleName.Has(rule.Name) {
```

Comment:
Didn't see any name is inserted to uniqueRuleName, how does it check uniqueness?